llvm/lib/Analysis/MLInlineAdvisor.cpp

```diff
@@ -206,23 +206,18 @@ void MLInlineAdvisor::onPassEntry(LazyCallGraph::SCC *LastSCC) {
   // care about the nature of the Edge (call or ref). `FunctionLevels`-wise, we
   // record them at the same level as the original node (this is a choice, may
   // need revisiting).
-  NodeCount -= static_cast<int64_t>(NodesInLastSCC.size());
   while (!NodesInLastSCC.empty()) {
     const auto *N = *NodesInLastSCC.begin();
+    assert(!N->isDead());
```

Comment:
could you add a note in the comment blurb above the `while`, to the "The cgscc pass manager rules", saying "Nodes are never deleted"? (IIUC that's now the case, right?)

Reply:
added comment